data-prepper-api/src/test/java/org/opensearch/dataprepper/model/failures/DlqObjectTest.java

```diff
@@ -0,0 +1,267 @@
+            assertThat(string, is(notNullValue()));
+
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(ISO8601_FORMAT_STRING);  // Specify locale to determine human language and cultural norms used in translating that input string.
+            Instant actualTimestamp = LocalDateTime.parse(testObject.getTimestamp(), formatter)
+                .atZone(ZoneId.systemDefault().normalized())
+                .toInstant();
+
+            assertThat(actualTimestamp, is(lessThanOrEqualTo(Instant.now())));
+        }
+
+        @Test
+        public void test_get_version() {
+            final String string = testObject.getVersion();
+            assertThat(string, is(notNullValue()));
+            assertThat(string, is(equalTo("1")));
+        }
+    }
+
+
+    @Nested
+    class EqualsAndHashCodeAndToString {
+
+        private DlqObject testObject;
+
+        @BeforeEach
+        public void setup() {
+            testObject = DlqObject.builder()
+                .withPluginId(pluginId)
+                .withPluginName(pluginName)
+                .withPipelineName(pipelineName)
+                .withFailedData(failedData)
+                .build();
+        }
+
+        @Test
+        void test_equals_returns_false_for_null() {
+            assertThat(testObject.equals(null), is(equalTo(false)));
+        }
+
+        @Test
+        void test_equals_returns_false_for_other_class() {
+            assertThat(testObject.equals(randomUUID()), is(equalTo(false)));
+        }
+
+        @Test
+        void test_equals_on_same_instance_returns_true() {
+            assertThat(testObject.equals(testObject), is(equalTo(true)));
+        }
+
+        @Test
+        void test_equals_a_clone_of_the_same_instance_returns_true() {
+
+            final DlqObject otherTestObject = DlqObject.builder()
+                .withPluginId(pluginId)
+                .withPluginName(pluginName)
+                .withPipelineName(pipelineName)
+                .withFailedData(failedData)
+                .withTimestamp(testObject.getTimestamp())
+                .build();
+
+            assertThat(testObject.equals(otherTestObject), is(equalTo(true)));
+        }
+
+        @Test
+        void test_equals_returns_false_for_two_instances_with_different_values() throws InterruptedException {
+            TimeUnit.SECONDS.sleep(1); // To ensure that the timestamp is different
+
+            final DlqObject otherTestObject = DlqObject.builder()
+                .withPluginId(pluginId)
+                .withPluginName(pluginName)
+                .withPipelineName(pipelineName)
+                .withFailedData(failedData)
+                .build();
+
+            assertThat(testObject, is(not(equalTo(otherTestObject))
+            ));
+        }
+
+        @Test
+        void test_hash_codes_for_two_instances_have_different_values() throws InterruptedException {
+            TimeUnit.SECONDS.sleep(1); // To ensure that the timestamp is different
```

Comment:
Rather than delay the test (even by a millisecond), I think you could just as easily create a specific instant. You could even do `.withTimestamp(Instant.now().plusSeconds(1))` (if you take my suggestion above about using Instant).